tachyon/math/base/semigroups.h

```diff
@@ -184,6 +184,14 @@ class MultiplicativeSemigroup {
     return ret;
   }
 
+  constexpr static G ExpPowOfTwo(size_t log_n, const G& g) {
+    G ret = g;
+    for (size_t i = 0; i < log_n; ++i) {
+      ret.SquareInPlace();
+    }
+    return ret;
+  }
+
```

Comment:
We treat exponent as `uint32_t` and also in order to match with the `Pow()`, how about modifying this as follows?

```suggestion
  constexpr auto ExpPowOfTwo(uint32_t log_n) const {
    G g = *static_cast<const G*>(this);
    for (size_t i = 0; i < log_n; ++i) {
      g.SquareInPlace();
    }
    return g;
  }
```